### DISCUSSION TITLE
Best way to add a new shipping method ?

### DISCUSSION THREAD

**Discussion by MiguelRita** [openstore-ecommerce/OpenStore#240] [Q&A]
Hello, I'm having a situation when adding a new shipping method. 
Long story short, the cart_recalculatesummary is throwing the following error:

`CART - ERROR!! - No Security rights or function command.`

**How did I add the shipping method?**
- I've duplicated the shipping plugin, only changed the "plugin Ref" and "Default Name" fields in the plugin edit options. Also replicated the interface from the original shipping plugin (assembly + Namespace class)
The security roles "Administrators" and "Managers" are checked.

I need two shipping methods, pickup at store (new) and delivery (existing default).

Any light on this matter would be grateful.
Thank you!!
Miguel R.

**Comment by leedavi** [comment]
I done this before for some of my clients.... Did you use the UI?  

If you used the UI, what you have done all sounds correct.

The best way to get multiple shipping providers is to add a new provider in the UI "Admin>Extensions".  And then copy all the settings from the original shipping provider, changing the ref.

Basically what you did but through the UI.
